### PR TITLE
CAMEL-14607: support negative acknowledge in pulsar message receipt

### PR DESCRIPTION
**Pull request by connormcauliffe-toast**
Addresses [Camel-14607](https://issues.apache.org/jira/browse/CAMEL-14607)

**Motivation**
Recently the pulsar version got updated to 2.5.0. In the new version of Pulsar, the negative acknowledgement is now a supported operation and we want to add support for that in Camel as well. This replaces the UnsupportedOperationException with the now supported `negativeAcknowledge` call.

**Verifying the change**
A test has been added to the `PulsarConsumerAcknowledgementTest` class which verifies that the negativeAck call redelivers the message. The test which verified the negativeAck call threw an UnsupportedOperationException has also been removed. 

**Additional Change**
There are minor changes in other test files which just remove deprecated Camel testing features. These were simply done to remove the warnings during the mvn build.

### CODE REVIEW

**Comment by connormcauliffe-toast** [comment]
I just realized there is a negative Ack redeliver delay property that would probably be nice to be able to configure. It isn't listed in the [concept section](https://pulsar.apache.org/docs/en/next/concepts-messaging/#negative-acknowledgement) so I didn't know that it was there until I was doing some more local testing and nothing was getting redelivered. Apparently the default is a minute which makes it very surprising that the test passes.

File: components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/configuration/PulsarConfiguration.java [hunk 1]
```diff
@@ -51,6 +51,8 @@
     private boolean allowManualAcknowledgement;
     @UriParam(label = "consumer", defaultValue = "10000")
     private long ackTimeoutMillis = 10000;
+    @UriParam(label = "consumer", defaultValue = "60000000")
+    private long negativeAckRedeliveryDelayMicros = 60000000;
```
Comment (davsclaus) [inline components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/configuration/PulsarConfiguration.java]: Lets use millis for this option as we do that for all others. micros is soo little that the number is hard to understand, so change it to millis thanks

**Comment by oscerd** [comment]
I'll wait for the fix suggested here

**Comment by connormcauliffe-toast** [comment]
The failure is from a sourcecheck error for a file in camel-core. Should I address that or is it fine? Same goes for my other PR

**Comment by oscerd** [comment]
No problem. I'll fix that after merging.

**Comment by oscerd** [comment]
Merged on master. Thanks

**Pull request closed without merging**